### ISSUE TITLE
Epochs

### ISSUE THREAD

**Issue by tpankaj** [bddmodelcar/training#1]
Add epoch saves, epoch training and validation loss calculation, and logging of epoch training and validation loss to CSV file

**Comment by raunakdoesdev** [comment]
Sure I'll see if I can add that tonight.
